tests/functional/n/no/no_name_in_module.py

```diff
@@ -30,6 +30,11 @@
 except ImportError:
     something = None
 
+try:
+    from collections import something
+except ModuleNotFoundError:
```

Comment:
Could you recheck some of these tests? Some are non-sensical (such as this one) in that they never occur in actual runtime `python`.
Since `collections` can be found this will/should never raise a `ModuleNotFoundError` and only raise a `ImportError`. I think we shouldn't bloat our tests with unfeasible examples.

Reply:
Indeed, sorry about this one. I was not sure about the other ones so I left them for the time being.

Reply:
I think the others are good!